lib/cosmos/interfaces/protocols/stream_protocol.rb

```diff
@@ -105,10 +77,9 @@ def read
         packet_data.replace(packet_data[@discard_leading_bytes..-1]) if @discard_leading_bytes > 0
 
         # Return data based on final_receive_processing
-        packet_data = post_read_data(packet_data)
         if packet_data
           if packet_data.length > 0
-            return post_read_packet(Packet.new(nil, nil, :BIG_ENDIAN, nil, packet_data))
+            return packet_data
```

Comment:
The higher level read method in Interface is now creating the Packet so just return the data